app/poi_api/poi_service.py

```diff
@@ -0,0 +1,106 @@
+from app import firestore_db
+from flask import jsonify
+from .poi_api import POIApi
+from .poi_suggestion import POI_suggestions
+from .poi import POI
+from .poi_errors import POINotFoundError, InvalidPOISuggestionError
+from datetime import datetime
+
+
+class POI_Service(POIApi):
+    def get_all_POI(self):
+        try:
+            poi_ref = firestore_db.collection("POI")
+            all_poi = [doc.to_dict() for doc in poi_ref.stream()]
+            list_all_poi = []
+            for poi in all_poi:
+                poi_object = self._create_poi_object(poi)
+                list_all_poi.append(poi_object)
+            return list_all_poi
+        except Exception as e:
+            raise Exception(f"An error occured: {e}")

```

Comment:
Sorry what exception happens here? Generally you should let the service class throw the exception and handle it wherever it is being called. Also try to avoid this sort of "catch all" exceptions.

Reply:
Yeah I guess it shouldn't throw an error. I just caught incase there was one. I can remove if it is unnecessary.

Reply:
Sometimes it's better for the error to be thrown and crash the system if it's really unexpected. Then we can look into solving for it.